v1/plugin/plugin.go

```diff
@@ -328,5 +384,186 @@ func startPlugin(srv server, m *meta, p *pluginProxy) int {
 	go p.HeartbeatWatch()
 	// TODO(danielscottt): exit code
 	<-p.halt
-	return 0
+
+	return nil
+}
+
+// GetPluginType converts a pluginType to a string
+// describing what type of plugin this is
+func getPluginType(plType pluginType) string {
+	switch plType {
+	case collectorType:
+		return "collector"
+
+	case publisherType:
+		return "publisher"
+
+	case processorType:
+		return "processor"
+	}
+	return ""
+}
+
+// GetPluginType converts a pluginType to a string
+// as described in snap/control/plugin/plugin.go
+func getRPCType(i int) string {
+	if i == 0 {
+		return "NativeRPC (0)"
```

Comment:
Good point, but It seems more thorough to test both cases. If somehow we do end up with NativeRPC, this will be printed in the diagnostics so we will be aware of the error right away.